src/tracks/InstrumentTrack.cpp

```diff
@@ -138,24 +145,32 @@ InstrumentTrack::InstrumentTrack( TrackContainer* tc ) :
 
 	setName( tr( "Default preset" ) );
 
-	connect( &m_baseNoteModel, SIGNAL( dataChanged() ),
-			this, SLOT( updateBaseNote() ), Qt::DirectConnection );
-	connect( &m_pitchModel, SIGNAL( dataChanged() ),
-			this, SLOT( updatePitch() ), Qt::DirectConnection );
-	connect( &m_pitchRangeModel, SIGNAL( dataChanged() ),
-			this, SLOT( updatePitchRange() ), Qt::DirectConnection );
-	connect( &m_effectChannelModel, SIGNAL( dataChanged() ),
-			this, SLOT( updateEffectChannel() ), Qt::DirectConnection );
+	connect(&m_baseNoteModel, SIGNAL(dataChanged()), this, SLOT(updateBaseNote()), Qt::DirectConnection);
+	connect(&m_pitchModel, SIGNAL(dataChanged()), this, SLOT(updatePitch()), Qt::DirectConnection);
+	connect(&m_pitchRangeModel, SIGNAL(dataChanged()), this, SLOT(updatePitchRange()), Qt::DirectConnection);
+	connect(&m_effectChannelModel, SIGNAL(dataChanged()), this, SLOT(updateEffectChannel()), Qt::DirectConnection);
 }
 
 
 int InstrumentTrack::baseNote() const
 {
 	int mp = m_useMasterPitchModel.value() ? Engine::getSong()->masterPitch() : 0;
 
-	return m_baseNoteModel.value() - mp;
+//	return m_microtuner.baseKey() - mp;
+	return DefaultBaseKey - mp;
 }
 
+int InstrumentTrack::firstKey() const
+{
+//	return m_microtuner.firstKey();
+	return 0;
```

Comment:
Why don't you return `m_firstKeyModel.value()`? If you're not going to use this for now, you can also remove related code from this PR.

Reply:
The first and last key models are needed only for the microtuner, so I tried to avoid using them while keeping most of the other code in place to avoid possible spaghetti moments (one removed function → another change required in a different file → another one elsewhere → before you know it it's a mess that does not work).

I did not realize that the first / last key models actually exist in this code, so returning a constant was a fast and safe thing to do. But looking at it now, when I return the model value, the range limits in PianoView actually work (the only thing missing is the disabled key highlighting). So I will fix that and keep it, since it actually provides a useful feature on its own.